public/js/controllers/proposal_controller.js

```diff
@@ -1,78 +1,180 @@
 import { Controller } from 'stimulus'
 import { getDefault } from '../helpers/module_helper'
+import { multiColumnBarPlotter, synchronize } from '../helpers/chart_helper'
+import dompurify from 'dompurify'
+import axios from 'axios'
 
+let common = {
+  labelsKMB: true,
+  legend: 'always',
+  fillGraph: true,
+  strokeWidth: 2,
+  labelsUTC: true,
+  gridLineColor: '#C4CBD2',
+  legendFormatter: legendFormatter
+}
+
+let percentConfig = {
+  ...common,
+  labels: ['Date', 'Percent Yes'],
+  ylabel: 'Approval (%)',
+  colors: ['#2971FF'],
+  labelsSeparateLines: true,
+  labelsDiv: 'percent-of-votes-legend',
+  underlayCallback: function (context, area, g) {
+    let xVals = g.xAxisExtremes()
+    let xl = g.toDomCoords(xVals[0], 60)
+    let xr = g.toDomCoords(xVals[1], 60)
+
+    context.beginPath()
+    context.strokeStyle = '#ED6D47'
+    context.moveTo(xl[0], xl[1])
+    context.lineTo(xr[0], xr[1])
+    context.fillStyle = '#ED6D47'
+    context.fillText('60% Approval', xl[0] + 20, xl[1] + 10)
+    context.closePath()
+    context.stroke()
+  }
+}
+
+let cumulativeConfig = {
+  ...common,
+  labels: ['Date', 'Total Votes'],
+  ylabel: 'Vote Count',
+  colors: ['#2971FF'],
+  labelsDiv: 'cumulative-legend',
+  underlayCallback: function (context, area, g) {
+    let xVals = g.xAxisExtremes()
+    let xl = g.toDomCoords(xVals[0], 8269)
+    let xr = g.toDomCoords(xVals[1], 8269)
+
+    context.beginPath()
+    context.strokeStyle = '#ED6D47'
+    context.moveTo(xl[0], xl[1])
+    context.lineTo(xr[0], xr[1])
+    context.fillStyle = '#ED6D47'
+    context.fillText('20% Quorum', xl[0] + 20, xl[1] + 10)
+    context.closePath()
+    context.stroke()
+  }
+}
+
+function legendFormatter (data) {
+  let html
+  if (data.x == null) {
+    html = data.series.map(function (series) {
+      return series.dashHTML + ' <span style="color:' +
+        series.color + ';">' + series.labelHTML + ' </span>'
+    }).join('<br>')
+  } else {
+    html = this.getLabels()[0] + ': ' + data.xHTML + ' UTC <br>'
+    data.series.forEach(function (series, index) {
+      if (!series.isVisible) return
+
+      var symbol = ''
+      if (series.labelHTML.includes('Percent')) {
+        series.labelHTML = 'Yes'
+        symbol = '%'
+      } else if (index === 0) {
+        html = ''
+      } else {
+        html += ' <br> '
+      }
+
+      var labeledData = '<span style="color:' + series.color + ';">' +
+        series.labelHTML + ' </span> : ' + Math.abs(series.y)
+      html += series.dashHTML + ' ' + labeledData + '' + symbol
+    })
+  }
+  dompurify.sanitize(html, { IN_PLACE: true })
```

Comment:
I think I gave you a bad example. `dompurify` is a little particular about certain inputs, but in your case, you don't need the second argument. ~~`, { IN_PLACE: true }`~~. Causing JS error.